sagemathinc/cocalc
# make pyutil fit for the world of python 3

# Description
* the goal is to make pyutil work with python3, in particular when there is only one global `/usr/bin/python` interpreter at version 3
* it also renames the commands yet again. `cc` looks like c-compiler and hard to discover. why not prefix it with cocalc to avoid any misunderstandings...

# Testing Steps
* for python3, I did `pip3 install --user -U ./`, etc.
* I basically checked if the text output looks reasonable or if the supposed conversion is happening

| utility                     | python2        | python3        | "test" | note |
| --------------------------- | -------------- | -------------- | --- | --- |
| cocalc-close                | :green_circle: | :green_circle: |
| cocalc-compute              | :green_circle: | :green_circle: | | cc-in-cc: start/stop/restart project, including sagews |
| cocalc-docx2txt             |   :green_circle: | :green_circle: |
| cocalc-html2sagews          | :green_circle: | :green_circle: |
| cocalc-ipynb-to-pdf         |  :green_circle: | :green_circle: |
| cocalc-ipynb2sagews         |   :green_circle: | :green_circle: |
| cocalc-jupyter              | :green_circle: |        :green_circle:     | :green_circle: | needs test in "test"
| cocalc-jupyter-classic-open | :green_circle: | :green_circle: |
| cocalc-jupyter-no-output    | :green_circle: | :green_circle: |
| cocalc-jupyterlab           | :green_circle: |       :green_circle:       | :green_circle: | needs test in "test"
| cocalc-ls                   |  :wastebasket:  | :wastebasket:  | | no longer used |
| cocalc-m2sagews             |   :wastebasket:  | :wastebasket: | | we have direct octave support |
| cocalc-new-file             | :green_circle: | :green_circle: | :green_circle: |    needs test in "test"
| cocalc-open                 | :green_circle: | :green_circle: | :green_circle: | needs test in "test"
| cocalc-rmd2html             | :wastebasket:  | :wastebasket:  | | we support rmd in a better way |
| cocalc-sagews2ipynb         |  :green_circle: | :green_circle: |
| cocalc-sagews2pdf           |  :green_circle: | :green_circle: | | 3d plots do not work |
| cocalc-start                |                |                | |  implicitly part of start/stop project |
| cocalc-status               | :green_circle: | :green_circle: | :green_circle: |
| cocalc-stop                 |                |                | | implicitly part of start/stop project |
| cocalc-sws2sagews           |      :green_circle: | :green_circle: |
| cocalc-top                  | :green_circle: | :green_circle: |

# Relevant Issues
also #4665

# Deployment
The corresponding kucalc branch is `py3-pyutil`

### [Checklist](https://github.com/sagemathinc/cocalc/wiki/PR-Checklist):

Front end:
